### PR TITLE
support for encoder from gemma-3-4b-it

### PR DESCRIPTION
**Pull request by vidixha**
This PR adds support for using the `gemma3-siglip-encoder` (from `google/gemma-3-4b-it`) as a vision tower for LLaVA pre-training with a Vicuna-based LLM.

**1. Numerical Instability issue - NaN loss**

Initial attempts to pre-train using the `gemma3-siglip-encoder` resulted in a persistent `NaN` loss. Debugging revealed that the encoder produces feature outputs with an extremely large numerical magnitude. This triggered a low-level bug deep inside the language model's `CrossEntropyLoss` function, causing it to fail even when all inputs (`logits` and `labels`) were valid.

**2. Implementation Details**
To enable stable training, the following two-part solution was implemented:

* **Feature Clipping:** A `torch.clamp` function was added to the `encode_images` method in `llava/model/llava_arch.py`. This controls the extreme magnitude of the `gemma` features by ensuring they are within a stable `[-10, 10]` range before being passed to the language model.

* **Manual Loss Calculation:** The `compute_loss` method in `llava/train/llava_trainer.py` was overridden to bypass the model's unstable internal loss function. This implementation takes the clean `logits` from the model and performs a stable, manual `CrossEntropyLoss` calculation.

### CODE REVIEW

File: llava/train/llava_trainer.py [hunk 1]
```diff
@@ -133,6 +133,60 @@ def __iter__(self):
 
 class LLaVATrainer(Trainer):
 
+    def compute_loss(self, model, inputs, return_outputs=False, **kwargs):
```
Comment (nahidalam) [inline llava/train/llava_trainer.py:136]: can we also make sure this `compute_loss` is bypassed in case of other encoders. I am still not sure why exactly we need a custom loss here. Is it to handle sequence mismatch problem? If so why is it not a problem in other encoders? Or is there any other reason? So unless those are clear, lets make sure we do this custom `compute_loss` only for gemma3-siglip
